packages/thirdweb/src/contract/deployment/deploy-with-abi.ts

```diff
@@ -160,21 +168,35 @@ export async function deployContract(
         chain: options.chain,
         client: options.client,
         to: info.create2FactoryAddress,
-        data: info.initBytecodeWithsalt,
+        data: info.initCalldata,
       }),
     });
-    return address;
+  } else {
+    const deployTx = prepareDirectDeployTransaction(options);
+    const receipt = await sendAndConfirmTransaction({
+      account: options.account,
+      transaction: deployTx,
+    });
+    address = receipt.contractAddress;
+    if (!address) {
+      throw new Error(
+        `Could not find deployed contract address in transaction: ${receipt.transactionHash}`,
+      );
+    }
   }
 
-  const deployTx = prepareDirectDeployTransaction(options);
-  const receipt = await sendAndConfirmTransaction({
-    account: options.account,
-    transaction: deployTx,
-  });
-  if (!receipt.contractAddress) {
-    throw new Error(
-      `Could not find deployed contract address in transaction: ${receipt.transactionHash}`,
-    );
+  if (options.isStylus) {
+    const activationTransaction = activateStylusContract({
+      chain: options.chain,
+      client: options.client,
+      contractAddress: address,
+    });
+
+    await sendTransaction({
```

Comment:
do we need to wait for confirmation or no?

Reply:
I think we should, at least to log the outcome of this.